src/lib.rs

```diff
@@ -134,7 +134,7 @@ macro_rules! nonempty {
     ($h:expr) => {
         $crate::NonEmpty {
             head: $h,
-            tail: alloc::vec::Vec::new(),
+            tail: vec![],
```

Comment:
This runs into issues with macro hygiene:
* https://doc.rust-lang.org/reference/macros-by-example.html#hygiene
* https://veykril.github.io/tlborm/decl-macros/minutiae/hygiene.html

It would have to be something like `$crate::__macro_support::vec` instead.

See e.g. https://github.com/tokio-rs/tracing/pull/1918